pkg/drain/nodegroup.go

```diff
@@ -143,6 +145,12 @@ func (n *NodeGroupDrainer) Drain() error {
 				if pending == 0 {
 					drainedNodes.Insert(node)
 				}
+
+				//only wait if we're not on the last node of this interation
+				if i < newPendingNodes.Len()-1 {
+					logger.Debug("Waiting for %.0f seconds before draining next node", n.NodeDrainWaitPeriod.Seconds())
+					time.Sleep(n.NodeDrainWaitPeriod)
```

Comment:
Hmm, is there a more effective wait to do this? What if it finished in between? Can we poll something to get the state? And if it's only on the last node, do we need to do this in the for loop? Can't we just create a value and then check the state of that node with a proper `select` sleep and channel using `time.After`?

Reply:
@Skarlso Happy to collab on this - I'm fairly new to Go so I'm sure there are slicker ways to handle this. This PR is part of my wanting to dig in and learn with a tool we're using. :)

Reply:
Okay, so we want to wait between nodes when they are being drained. And we want to wait until they are done draining, right? So, what I'm suggesting is using a ticker https://gobyexample.com/tickers. But we have to find out how we can check if a node is drained or not.

I guess this check `evictPods` will return pods that are still being evicted. So instead of `continue`-ing when this is coming back as not `0`, let's do a ticker wait continuously asking if there are still pods to evict before moving on to the next node. I think this is the desired behaviour, right?